test/test_contrib.py

```diff
@@ -436,6 +436,21 @@ def test_gradcheck(self, device, dtype):
         input1 = torch.ones(B, C, H, W, device=device, dtype=dtype, requires_grad=True)
         assert gradcheck(kornia.contrib.distance_transform, (input1), raise_exception=True)
 
+    def test_loss_grad(self, device, dtype):
+        B, C, H, W = 1, 1, 32, 32
+        input1 = torch.rand(B, C, H, W, device=device, dtype=dtype, requires_grad=True)
+        input2 = torch.rand(B, C, H, W, device=device, dtype=dtype, requires_grad=True)
+        tiny_module = nn.Conv2d(1, 1, (3, 3), (1, 1), (1, 1))
+        input1 = kornia.contrib.distance_transform(tiny_module(input1))
+        input2 = kornia.contrib.distance_transform(input2)
+        loss = torch.nn.functional.mse_loss(input1, input2)
+        try:
```

Comment:
```suggestion
        assert gradcheck(loss, (input1, input2,), raise_exception=True)
```
I believe this should work, or similar

Reply:
And you might want to lower input sizes if you see it’s slow

Reply:
`gradcheck(mse_loss, (input1, input2,), raise_exception=True)` won't be able to find the bug during `loss.backward()`.
I think that's why this bug exists even when there's a grad check at https://github.com/kornia/kornia/blob/fc4b2fb7e7b0a8abbc802e67e8406a5a32c32d62/test/test_contrib.py#L434-L437

Reply:
I see. In that case, gradcheck only check the gradients computation but not check if we can backward or not. So that there might be other non-backwardable functions in the lib.

Reply:
Very interesting to see that since we rely quite a lot on `gradcheck` to verify differentiability. /cc @shijianjian @ducha-aiki @lferraz

Reply:
I guess you are removing the `try/catch` block after the fix

Reply:
The `try/catch` block is removed in https://github.com/kornia/kornia/pull/1584/commits/dcd1c5e17cf4d2ae2db1f438c53245bba0afd93f .
Up to now I'm not aware of a better way to check whether gradient could be back-propagated. 
Any suggestions?